processing/src/main/java/io/druid/query/filter/BoundDimFilter.java

```diff
@@ -53,6 +56,19 @@ public BoundDimFilter(
     this.lowerStrict = (lowerStrict == null) ? false : lowerStrict;
     this.upperStrict = (upperStrict == null) ? false : upperStrict;
     this.alphaNumeric = (alphaNumeric == null) ? false : alphaNumeric;
+    this.extractionFn = extractionFn;
+  }
+
+  public BoundDimFilter(
```

Comment:
IMO one nice thing about only having one constructor is that it makes it harder to accidentally create an object with a critical parameter missing.